metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetector.java

```diff
@@ -1,20 +1,174 @@
 
-  @Override
-  protected void doStop() {}
+  static final String TENANT_ID_KEY = "x-tenant-id";
 
-  @Override
-  public boolean healthCheck() {
-    return false;
+  private final MetricQueryBuilder metricQueryBuilder;
+  private final QueryServiceClient queryServiceClient;
+  private final int qsRequestTimeout;
+
+  MetricAnomalyDetector(Config appConfig) {
+    AttributeServiceClientConfig asConfig = AttributeServiceClientConfig.from(appConfig);
+    ManagedChannel attributeServiceChannel =
+        ManagedChannelBuilder.forAddress(asConfig.getHost(), asConfig.getPort())
+            .usePlaintext()
+            .build();
+    AttributeServiceClient asClient = new AttributeServiceClient(attributeServiceChannel);
+
+    Config qsConfig = appConfig.getConfig(QUERY_SERVICE_CONFIG_KEY);
+    queryServiceClient = new QueryServiceClient(new QueryServiceConfig(qsConfig));
+    qsRequestTimeout =
+        appConfig.hasPath(REQUEST_TIMEOUT_CONFIG_KEY)
+            ? appConfig.getInt(REQUEST_TIMEOUT_CONFIG_KEY)
+            : DEFAULT_REQUEST_TIMEOUT_MILLIS;
+
+    metricQueryBuilder = new MetricQueryBuilder(asClient);
+  }
+
+  void process(AlertTask alertTask) {
+    MetricAnomalyEventCondition metricAnomalyEventCondition;
+    if (alertTask.getEventConditionType().equals(METRIC_ANOMALY_EVENT_CONDITION)) {
+      try {
+        metricAnomalyEventCondition =
+            MetricAnomalyEventCondition.parseFrom(alertTask.getEventConditionValue());
+      } catch (InvalidProtocolBufferException e) {
+        LOGGER.error("Exception while parsing event condition", e);
+        return;
+      }
+    } else {
+      LOGGER.debug("EventConditionType:{}", alertTask.getEventConditionType());
+      return;
+    }
+
+    QueryRequest queryRequest =
+        metricQueryBuilder.buildMetricQueryRequest(
+            metricAnomalyEventCondition.getMetricSelection(),
+            alertTask.getLastExecutionTime(),
+            alertTask.getCurrentExecutionTime(),
+            alertTask.getTenantId());
+
+    // todo handle multiple violation conditions
+    ViolationCondition violationCondition =
+        metricAnomalyEventCondition.getViolationConditionList().get(0);
+
+    Iterator<ResultSetChunk> iterator =
+        executeQuery(Map.of(TENANT_ID_KEY, alertTask.getTenantId()), queryRequest);
+
+    LOGGER.debug(
+        "Starting rule evaluation for rule Id {} start {} & end time {}",
+        alertTask.getEventConditionId(),
+        Instant.ofEpochMilli(alertTask.getLastExecutionTime()),
+        Instant.ofEpochMilli(alertTask.getCurrentExecutionTime()));
+
+    if (violationCondition.hasStaticThresholdCondition()) {
+      evaluateForStaticThreshold(
+          iterator,
+          violationCondition,
+          metricAnomalyEventCondition.getMetricSelection().getMetricAttribute());
+    }
+  }
+
+  void evaluateForStaticThreshold(
+      Iterator<ResultSetChunk> iterator,
```

Comment:
Note: as a follow-up, this should take an internal TimeSeries structure. So, 
- metric data fetcher is isolated from the evaluator
- it will be needed while taking care of the condition last 5 mins as we have to maintain the in-memory state
- we may also have to take care of holes if data points are missing

Can you create a task if it's not there - https://github.com/hypertrace/hypertrace/issues/251 ?

Reply:
Done https://github.com/hypertrace/hypertrace/issues/263